src/core/cgroup.c

```diff
@@ -2785,7 +2785,7 @@ int unit_check_oomd_kill(Unit *u) {
         if (!increased)
                 return 0;
 
-        if (n > 0)
+        if (n > 0 && !unit_filters_log_level(u, LOG_NOTICE))
                 log_struct(LOG_NOTICE,
```

Comment:
hmm, please instead add log_unit_struct() or so, that adds LOG_UNIT_ID(u) and so on implicitly, and does this check internally, too

Reply:
`LOG_UNIT_ID(u)` (etc.) doesn't appear in exactly the same place in every such call. There's a pattern I can abstract that covers the majority of the cases; should I abstract that and leave the remainder as a combination of `unit_log_level_test` and `log_struct`, or have `log_unit_struct` not include `LOG_UNIT_ID(u)` at all?

Reply:
the order of the fields doesn't matter, journald might reorder anyway

Reply:
Ah, I see, thanks!